FWCore/Framework/interface/EDConsumerBase.h

```diff
@@ -267,6 +294,59 @@ namespace edm {
     bool frozen_;
     bool containsCurrentProcessAlias_;
   };
+
+  template <Transition TR>
+  class EDConsumerBaseESAdaptor {
+  public:
+    template <typename TYPE, typename REC>
+    ESGetToken<TYPE, REC> consumes() const {
+      return m_consumer->template esConsumes<TYPE, REC, TR>();
+    }
+
+  private:
+    //only EDConsumerBase is allowed to make an instance of this class
+    friend class EDConsumerBase;
+    EDConsumerBaseESAdaptor(EDConsumerBase* iBase) : m_consumer(iBase) {}
+
+    EDConsumerBase* m_consumer;
+  };
+
+  template <Transition TR>
+  class EDConsumerBaseWithTagESAdaptor {
+  public:
+    template <typename TYPE, typename REC>
+    ESGetToken<TYPE, REC> consumes() const {
+      return m_consumer->template esConsumes<TYPE, REC, TR>(m_tag);
+    }
+
+  private:
+    //only EDConsumerBase is allowed to make an instance of this class
+    friend class EDConsumerBase;
+    EDConsumerBaseWithTagESAdaptor(EDConsumerBase* iBase, ESInputTag iTag) noexcept
+        : m_consumer(iBase), m_tag(std::move(iTag)) {}
+
+    EDConsumerBase* m_consumer;
+    ESInputTag const m_tag;
+  };
+
+  template <BranchType B>
+  class EDConsumerBaseAdaptor {
+  public:
+    template <typename TYPE>
+    EDGetTokenT<TYPE> consumes() const {
```

Comment:
```suggestion
    EDGetTokenT<TYPE> consumes() {
```
assuming the function would better be non-`const`.